modules/arm_plugin/demos/coco_detection_android_demo/app/src/main/java/org/intel/openvino/MainActivity.java

```diff
@@ -0,0 +1,273 @@
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        if (grantResults.length > 0 && grantResults[0] != PackageManager.PERMISSION_GRANTED) {
+            Log.e("PermissionError", "The application can't work without camera permissions");
+            System.exit(1);
+        }
+        processNetwork();
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        mOpenCvCameraView.enableView();
+    }
+
+    @Override
+    public void onCameraViewStarted(int width, int height) {}
+
+    @Override
+    public void onCameraViewStopped() {}
+
+    @Override
+    public Mat onCameraFrame(CvCameraViewFrame inputFrame) {
+        // Use TickMeter to calculate fps
+        tm.start();
+        Mat frame = inputFrame.rgba();
+        Imgproc.cvtColor(frame, frame, Imgproc.COLOR_RGBA2RGB);
+        Mat currentFrame = frame.clone();
+        frame.release();
+
+        // Preprocess the frame
+        Imgproc.resize(currentFrame, currentFrame, new Size(300, 300));
+        int[] dimsArr = {1, currentFrame.rows(), currentFrame.cols(), 3};
+        Tensor input_tensor = new Tensor(ElementType.u8, dimsArr, currentFrame.dataAddr());
+
+        // Set input data
+        inferRequest.set_input_tensor(input_tensor);
+        inferRequest.infer();
+
+        // Get output data from model
+        Tensor output_tensor = inferRequest.get_output_tensor();
+        Imgproc.resize(currentFrame, currentFrame, new Size(320, 240));
+
+        float[] detection = output_tensor.data();
+        int maxProposalCount = detection.length / 7;
+
+        // Construct the input to the NMS algorithm
+        List<Rect2d> rect2dList = new ArrayList<>();
+        List<Float> confList = new ArrayList<>();
+        List<Integer> objIndexList = new ArrayList<>();
+
+        for (int i = 0; i < maxProposalCount; i ++) {
+            float label = detection[i * 7 + 1];
+            float conf = detection[i * 7 + 2];
+            float xMin = detection[i * 7 + 3] * currentFrame.cols();
+            float yMin = detection[i * 7 + 4] * currentFrame.rows();
+            float xMax = detection[i * 7 + 5] * currentFrame.cols();
+            float yMax = detection[i * 7 + 6] * currentFrame.rows();
+
+            confList.add(conf);
+            objIndexList.add((int) label);
+            rect2dList.add(new Rect2d(xMin, yMin, (xMax - xMin), (yMax - yMin)));
+        }
+
+        MatOfInt indices = new MatOfInt();
+        MatOfRect2d boxes = new MatOfRect2d(rect2dList.toArray(new Rect2d[0]));
+        float[] confArr = new float[confList.size()];
+        for (int i = 0; i < confList.size(); i++) {
+            confArr[i] = confList.get(i);
+        }
+        MatOfFloat confs = new MatOfFloat(confArr);
+        Dnn.NMSBoxes(boxes, confs, CONFIDENCE_THRESHOLD, NMS_THRESHOLD, indices);
```

Comment:
SSD MobileNet already applies NMS internally on the last layer. Does this method change results in your experiments?

Reply:
Refer to this notebook: https://github.com/openvinotoolkit/openvino_notebooks/blob/main/notebooks/401-object-detection-webcam/401-object-detection.ipynb.

I tried some other models, like yolo, which does not contains NMS function, so I added this function for adapting other models, but it has no major impact on SSD MobileNet.

<img width="731" alt="image" src="https://user-images.githubusercontent.com/47499836/189607822-3a6d6b4b-44cb-4933-8725-c18a2dfd1966.png">

Reply:
That's right, only YOLO with multiple outputs require extra NMS (like YOLOv3 and newer). Ok, let's keep NMS in the sample at least for a demonstration of how to use it. Anyway OpenCV is used.